### PR TITLE
IGNITE-14403 Added ignite-core module and IgniteUuid class

### PR DESCRIPTION
**Pull request by agura**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/lang/IgniteUuid.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+import java.io.Serializable;
+import java.util.UUID;
+
+/**
+ * This is a faster performing version of {@link UUID}. On basic tests this version is at least
+ * 10x time faster for ID creation. It uses extra memory for 8-byte counter additionally to
+ * internal UUID.
+ */
+public final class IgniteUuid implements Comparable<IgniteUuid>, Cloneable, Serializable {
```
Comment (AMashenkov) [inline modules/core/src/main/java/org/apache/ignite/lang/IgniteUuid.java:28]: Let's describe IgniteUuid natural order here.

File: pom.xml [hunk 2]
```diff
@@ -34,6 +34,7 @@
     <packaging>pom</packaging>
 
     <modules>
+        <module>modules/core</module>
```
Comment (AMashenkov) [inline pom.xml:37]: Let's add a new module respecting alphabetic order.

File: modules/core/src/main/java/org/apache/ignite/lang/IgniteUuidGenerator.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+        cntGen = new AtomicLong(genStart);
+    }
+
+    /**
+     * Gets {@link UUID} associated with local VM.
+     *
+     * @return {@link UUID} associated with local VM.
+     */
+    public UUID vmId() {
```
Comment (AMashenkov) [inline modules/core/src/main/java/org/apache/ignite/lang/IgniteUuidGenerator.java:51]: I'm not sure if "vmID" is a suitable name here.
What does VM means?

Comment (agura) [reply modules/core/src/main/java/org/apache/ignite/lang/IgniteUuidGenerator.java:51]: @AMashenkov See javadoc:

>{@link UUID} associated with **local VM**.

**Review by ibessonov — APPROVED** [review]

File: pom.xml (hunk 2 above)
Comment (agura) [reply pom.xml:37]: Okay, thanks!

File: modules/core/src/main/java/org/apache/ignite/lang/IgniteUuid.java (hunk 1 above)
Comment (agura) [reply modules/core/src/main/java/org/apache/ignite/lang/IgniteUuid.java:28]: Not a scope. Could you please describe it during storage related work?

**Pull request merged**